### PR TITLE
Plus menu component

### PR DESCRIPTION
**Pull request by Hypercubed**
## Summary

* Adds a plus-menu component

## Checklist

- [ ] \*Added unit tests
- [x] \*Added a code reviewer
- [x] Added changes to `/projects/swimlane/ngx-ui/CHANGELOG.md` under HEAD (Unreleased)
- [x] Updated the demo page
- [ ] Included screenshots of visual changes

_\*required_

image

image

### CODE REVIEW

File: projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+@Component({
+  selector: 'ngx-plus-menu',
+  templateUrl: './plus-menu.component.html',
+  styleUrls: ['./plus-menu.component.scss'],
+  host: { class: 'ngx-plus-menu' },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None
+})
+export class PlusMenuComponent {
+  @Input() items = [];
```
Comment (haeminn) [inline projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.ts:23]: Can we type these?

File: src/app/components/plus-menu-page/plus-menu-page.component.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    display: flex;
+    justify-content: center;
+
+    ngx-plus-menu {
+      top: -20px;
+    }
+  }
+
+  // ICON corrections
+  // .ngx-icon.ngx-add-circle-thin {
```
Comment (haeminn) [inline src/app/components/plus-menu-page/plus-menu-page.component.scss:43]: comments

File: projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.ts (hunk 1 above)
Comment (Hypercubed) [reply projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.ts:23]: yes

File: projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.ts [hunk 3]
```diff
@@ -20,7 +27,7 @@ import { PlusMenuPosition } from './plus-menu-position.enum';
   encapsulation: ViewEncapsulation.None
 })
 export class PlusMenuComponent {
-  @Input() items = [];
+  @Input() items: PlusMenuItem[] = [];
   @Input() position = PlusMenuPosition.Right;
```
Comment (haeminn) [inline projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.ts:31]: sorry, i meant typing all inputs

Comment (Hypercubed) [reply projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.ts:31]: Like @Input() menuColor = '#9fce36' not sure we need to add obvious types like that.

**Review by haeminn — APPROVED** [review]

**Review by haeminn — APPROVED** [review]
lgtm

**Review by nelsonGuti — APPROVED** [review]
Looks Good, Great work!

**Review by nartc — APPROVED** [review]

File: projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.scss [hunk 4]
```diff
@@ -0,0 +1,551 @@
+@import 'colors/variables';
+
+.ngx-plus-menu {
+  --menu-color: #9fce36;
+  --background-color: 0, 0, 0;
```
Comment (mcwenger) [inline projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.scss:5]: Let's make --background-color `$color-blue-grey-900`

File: projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.scss [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+    color: white;
+
+    .ngx-plus-menu--item {
+      position: absolute;
+      width: 90px;
+      height: auto;
+      text-align: right;
+      font-size: 18px;
+      line-height: 19px;
+      font-weight: bold;
```
Comment (mcwenger) [inline projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.scss:96]: Weight should be semibold. It's `600` for Google fonts, such as Source Sans Pro.

File: projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.scss [hunk 6]
```diff
@@ -0,0 +1,551 @@
...
+      cursor: pointer;
+
+      .keys {
+        font-style: normal;
+        font-weight: 600;
+        font-size: 12px;
```
Comment (mcwenger) [inline projects/swimlane/ngx-ui/src/lib/components/plus-menu/plus-menu.component.scss:102]: Let's bump the font size to 13px. There's a rendering difference between Figma and web fonts, and small fonts sometimes need to be bumped up—even though you implemented exactly as represented in design.

**Review by mcwenger — CHANGES_REQUESTED** [review]
Overall amazing—just a few small style adjustments. Thanks!

**Review by mcwenger — APPROVED** [review]
LGTM!

**Pull request merged**